validator/project_validator_test.go

```diff
@@ -1298,39 +1301,63 @@ func TestCheckProjectSemantics(t *testing.T) {
 	})
 }
 
-func TestEnsureHasNecessaryProjectFields(t *testing.T) {
-	Convey("When ensuring necessary project fields are set, ensure that", t, func() {
-		Convey("projects validate all necessary fields exist", func() {
-			Convey("an error should be thrown if the batch_time field is "+
-				"set to a negative value", func() {
-				project := &model.Project{
-					Enabled:     true,
-					Identifier:  "identifier",
-					Owner:       "owner",
-					Repo:        "repo",
-					Branch:      "branch",
-					DisplayName: "test",
-					RepoKind:    "github",
-					BatchTime:   -10,
-				}
-				So(ensureHasNecessaryProjectFields(project),
-					ShouldNotResemble, []ValidationError{})
-				So(len(ensureHasNecessaryProjectFields(project)),
-					ShouldEqual, 1)
-			})
-			Convey("an error should be thrown if the command type "+
-				"field is invalid", func() {
-				project := &model.Project{
-					BatchTime:   10,
-					CommandType: "random",
-				}
-				So(ensureHasNecessaryProjectFields(project),
-					ShouldNotResemble, []ValidationError{})
-				So(len(ensureHasNecessaryProjectFields(project)),
-					ShouldEqual, 1)
-			})
-		})
-	})
+type EnsureHasNecessaryProjectFieldSuite struct {
+	suite.Suite
+}
+
+func TestEnsureHasNecessaryProjectFieldSuite(t *testing.T) {
+	suite.Run(t, new(EnsureHasNecessaryProjectFieldSuite))
+}
+
+func (s *EnsureHasNecessaryProjectFieldSuite) TestBatchTimeValueMustNonNegative() {
+	assert := assert.New(s.T())
+	project := &model.Project{
+		Enabled:     true,
+		Identifier:  "identifier",
+		Owner:       "owner",
+		Repo:        "repo",
+		Branch:      "branch",
+		DisplayName: "test",
+		RepoKind:    "github",
+		BatchTime:   -10,
+	}
```

Comment:
I might use the SetupTest method/interface to define a project document on the suite that resets for each test.